### PR TITLE
[SPARKR-154] Phase 3: implement intersection().

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -1487,3 +1487,39 @@ setMethod("zipRDD",
...
+#' Intersection of this RDD and another one.
+#'
+#' Return the intersection of this RDD and another one.
+#' The output will not contain any duplicate elements, 
+#' even if the input RDDs did. Performs a hash partition
+#' across the cluster.
+#' Note that this method performs a shuffle internally. 
+#'
+#' @param x An RDD.
+#' @param other An RDD.
```
Comment (concretevitamin) [inline pkg/R/RDD.R]: Should we document the third argument `numPartitions`

**Comment by concretevitamin** [comment]
LGTM pending a minor doc issue.

File: pkg/R/RDD.R (hunk 1 above)
Comment (sun-rui) [inline pkg/R/RDD.R]: added.

**Comment by shivaram** [comment]
Thanks @sun-rui - LGTM. Merging this and cherry-picking to sparkr-sql. 

Could you also open this as a PR against amplab-extras/spark to keep things in sync ? Let me know if you have any trouble with that.

**Pull request merged**